src/wallet/init.cpp

```diff
@@ -46,7 +46,7 @@ void WalletInit::AddWalletOptions() const
 
     gArgs.AddArg("-fallbackfee=<amt>", strprintf("A fee rate (in %s/kB) that will be used when fee estimation has insufficient data. 0 to entirely disable the fallbackfee feature. (default: %s)",
                                                                CURRENCY_UNIT, FormatMoney(DEFAULT_FALLBACK_FEE)), ArgsManager::ALLOW_ANY, OptionsCategory::WALLET);
-    gArgs.AddArg("-keypool=<n>", strprintf("Set key pool size to <n> (default: %u)", DEFAULT_KEYPOOL_SIZE), ArgsManager::ALLOW_ANY, OptionsCategory::WALLET);
+    gArgs.AddArg("-keypool=<n>", strprintf("Set key pool size to <n> (default: %u). Decreasing keypool size to a low value may increase the risk of losing funds in case of restoring from an old backup if none of the addresses in the original keypool have been used.", DEFAULT_KEYPOOL_SIZE), ArgsManager::ALLOW_ANY, OptionsCategory::WALLET);
```

Comment:
Consider:
```diff
-    gArgs.AddArg("-keypool=<n>", strprintf("Set key pool size to <n> (default: %u). Decreasing keypool size to a low value may increase the risk of losing funds in case of restoring from an old backup if none of the addresses in the original keypool have been used.", DEFAULT_KEYPOOL_SIZE), ArgsManager::ALLOW_ANY, OptionsCategory::WALLET);
+    gArgs.AddArg("-keypool=<n>", strprintf("Set key pool size to <n> (default: %u). "
+                                           "Smaller sizes may increase the risk of losing funds when restoring from an "
+                                           "old backup, if none of the addresses in the original keypool have been used.",
+                                           DEFAULT_KEYPOOL_SIZE), ArgsManager::ALLOW_ANY, OptionsCategory::WALLET);
```

Reply:
maybe want to clarify that the original keypool was larger

Reply:
My understanding is if none of the addresses is detected as used on-chain, the keypool is not going to be top-up. If size is N keypool stays at range 0..N instead of being shifted by I, with I being index of address detected. Size is always going to be constant ?

Or do you mean original keypool as keypool set by default ?

Reply:
@ariard who are you asking here? sorry

Reply:
@instagibbs what your comment that seems unclear to me

Reply:
I'm saying that the explanation should explicitly mention that the value for the "original keypool" was larger in this loss-risk case.

Reply:
The original keypool as set by a mindless user before any detection in the loss-risk case or the default keypool ?
First option doesn't make sense to me as AFAII keypool keeps the same size.